File: python/problem3.py
# problem3.py
def dividir(a: float, b: float):
    """
    Realiza la división a / b.
    """
    try:
        return a / b
    except ZeroDivisionError:
        return "Error: dividiste por cero"


if __name__ == "__main__":
    try:
        a = float(input("Numerador: "))
        b = float(input("Denominador: "))
        resultado = dividir(a, b)
        print("Resultado:", resultado)
    except ValueError:
        print("Error ingrese numeros validos.")

File: haskell/problem3.hs
-- Problem3.hs
module Problem3 (dividir) where

-- La funcion devuelve Either:
--   Right resultado si la division es valida
--   Left mensaje de error si el divisor es cero
dividir :: (Eq a, Fractional a) => a -> a -> Either String a
dividir _ 0 = Left "Error: division por cero"
dividir a b = Right (a / b)

-- funcion principal de prueba
main :: IO ()
main = do
    print (dividir 10 2)   -- Right 5.0
    print (dividir 7 0)    -- Left "Error: division por cero"
    print (dividir 5 2)    -- Right 2.5
